lib/cpp/src/thrift/concurrency/TimerManager.cpp

```diff
@@ -290,11 +292,23 @@ void TimerManager::add(shared_ptr<Runnable> task, const struct timeval& value) {
 }
 
 void TimerManager::remove(shared_ptr<Runnable> task) {
-  (void)task;
   Synchronized s(monitor_);
   if (state_ != TimerManager::STARTED) {
     throw IllegalStateException();
   }
+  std::vector<task_iterator> toRemove;
```

Comment:
I'd recommend using the prefix/postfix method to walk and erase here in order to avoid iterator invalidation:

    bool found = false;
    for (task_iterator it = taskMap_.begin(); it != taskMap_.end(); ) {
      if (ix->second->runnable() == task) {
        found = true;
        taskMap_.erase(it++);
      } else {
        ++it;
      }
    }

    if (!found) {
      throw NoSuchTaskException();
    }

Reply:
done